### PR TITLE
set (various attributes) should be nullable

### PR DESCRIPTION
**Pull request by taquitos**
Also make warnings errors in APITester project to catch related things like this.

### CODE REVIEW

**Comment by aboedo** [comment]
we should add a unit test that checks that sending in `nil` actually deletes the attribute

**Review by aboedo — COMMENTED** [review]
looking great so far! I think we should have a unit test, so that we ensure that the behavior is correct, not just the API

**Comment by taquitos** [comment]
@aboedo yup! I was just kicking this off with some initial commits to see if you had any other thoughts 😄 I was always going to follow up with unit tests. Speaking of.... we should definitely add to our PR template "I added unit tests". Any new code should be tested!

File: PurchasesTests/Purchasing/PurchasesTests.swift [hunk 1]
```diff
@@ -2016,6 +2016,89 @@ class PurchasesTests: XCTestCase {
         verifyUpdatedCaches(newAppUserID: identityManager.currentAppUserID)
     }
 
+    func testSetAndClearSubscriberAttributesPassesThroughToAttributesManager() {
```
Comment (aboedo) [inline PurchasesTests/Purchasing/PurchasesTests.swift:2019]: nitpick: I'd split these into separate cases so that if the test fails we immediately know which method was messed up

File: PurchasesTests/Purchasing/PurchasesTests.swift [hunk 2]
```diff
@@ -2016,6 +2016,89 @@ class PurchasesTests: XCTestCase {
         verifyUpdatedCaches(newAppUserID: identityManager.currentAppUserID)
     }
```
Comment (aboedo) [inline PurchasesTests/Purchasing/PurchasesTests.swift:2018]: since this file is pretty big, we added the tests for subscriber attributes stuff to a separate one, https://github.com/RevenueCat/purchases-ios/blob/main/PurchasesTests/SubscriberAttributes/PurchasesSubscriberAttributesTests.swift

We should move these to that file since they're testing related stuff

**Review by aboedo — APPROVED** [review]
looks great!

File: PurchasesTests/Purchasing/PurchasesTests.swift (hunk 2 above)
Comment (taquitos) [reply PurchasesTests/Purchasing/PurchasesTests.swift:2018]: ✅

File: PurchasesTests/Purchasing/PurchasesTests.swift (hunk 1 above)
Comment (taquitos) [reply PurchasesTests/Purchasing/PurchasesTests.swift:2019]: ✅
